mac

```diff
@@ -86,7 +86,6 @@ gem_install_or_update() {
     gem update "$@"
   else
     gem install "$@"
-    rbenv rehash
```

Comment:
Explicit `asdf reshim ruby` shouldn‘t be necessary — asdf-ruby automatically reshims on `Gem.post_install` (via a [rubygem plugin](https://github.com/asdf-vm/asdf-ruby/blob/master/rubygems-plugin/rubygems_plugin.rb)). I confirmed this by installing a fresh ruby version and setting that as the local version, noting the absence of the `rspec` shim with `ls ~/.asdf/shims/`, then did `gem install rspec` and indeed the shim `~/.asdf/shims/rspec` was automatically created.

Also, explicit `rbenv rehash` hasn‘t been needed under normal circumstances since the behavior of [rbenv-gem-rehash](https://github.com/rbenv/rbenv-gem-rehash) was [integrated into core rbenv](https://github.com/rbenv/rbenv/pull/638).

(The asdf-ruby rubygem plugin is an old version of [rbenv‘s rubygem-plugin](https://github.com/rbenv/rbenv/blob/master/rbenv.d/exec/gem-rehash/rubygems_plugin.rb). The current rbenv version has been enhanced to rehash on `Gem.post_uninstall`, avoid rehashing when no new executables were added by a Bundler gem install, and only runs rehash once per Bundle install rather than once per gem.)